web-api/src/storage/postgres.rs

```diff
@@ -105,45 +100,38 @@ impl Database {
 
     pub(super) async fn insert_documents(
         &self,
-        documents: impl IntoIterator<Item = &IngestedDocument>,
+        documents: &[DocumentForIngestion],
     ) -> Result<(), Error> {
         let mut tx = self.begin().await?;
 
         let mut builder = QueryBuilder::new(
             "INSERT INTO document (
                 document_id,
-                snippet,
+                raw_document,
```

Comment:
:pushpin: Depending how we will handle `raw_documents` we might want to consider returning a hash of the document instead (it's used for deciding if an upsert is a property update or full document change).